src/components/form/FormPersonalDetails.vue

```diff
@@ -0,0 +1,233 @@
+<script lang="ts">
+/**
+ * FormPersonalDetails Component
+ *
+ * The `FormPersonalDetails`
+ *
+ * @description * Use this component to display contact details form.
+ *
+ * Note: This component is commonly used in `RegisterChallengePage`.
+ *
+ * @props
+ * - `modelValue` (Object, required): Reactive object representing form state.
+ *
+ * @events
+ * - `update:modelValue`: Emitted as a part of v-model structure.
+ *
+ * @components
+ * - `FormFieldTextRequired`: Component to render name, surname, nickname...
+ * - `FormFieldRadioRequired`: Component to render gender radio buttons.
+ *
+ * @example
+ * <form-personal-details />
+ *
+ * @see [Figma Design](https://www.figma.com/file/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?type=design&node-id=4858%3A102976&mode=dev)
+ */
+
+import { defineComponent, nextTick } from 'vue';
+
+// components
+import FormFieldTextRequired from 'components/global/FormFieldTextRequired.vue';
+
+export type FormPersonalDetailsFields = {
```

Comment:
Shouldn't the types be in the types directory?

Reply:
Yes, they should. Corrected.